dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -484,11 +486,55 @@ private String getQuerySelect( TrackedEntityInstanceQueryParams params, boolean
         {
             select.append( ", TEI.trackedentityinstanceid AS teiid" );
         }
+
+        if ( params.getOrders() != null )
+        {
+            select.append( getOrderByQuery( params, select ) );
+        }
+
         select.append( SPACE );
 
         return select.toString();
     }
 
+    private String getOrderByQuery( TrackedEntityInstanceQueryParams params, StringBuilder select )
```

Comment:
This method name is confusing to me 🤔 If I understand the code correctly: you are adding select expressions for the columns that we will order by http://forcedotcom.github.io/phoenix/#select

There is already `getQueryOrderBy` which is actually generating the ORDER BY clause.

These two method names are too similar while doing different things.

Maybe we can call it `getQuerySelectOrderByColumns`, `getQuerySelectOrderBy`, ...